static/js/apps/nl_eval/app.tsx

```diff
@@ -15,12 +15,60 @@
  */
 
 import axios from "axios";
-import React, { useState } from "react";
+import _ from "lodash";
+import React, { useEffect, useRef, useState } from "react";
 
 import { stringifyFn } from "../../utils/axios";
 import { QuerySection } from "./query_section";
 import { EmbeddingObject } from "./util";
 
+function OverallScoreTable(props: {
+  data: Record<string, Record<string, number>>;
+  count: number;
+}): JSX.Element {
+  const [mark, setMark] = useState(null);
+  useEffect(() => {
+    // Check if data is complete. If so, set mark to an empty string so that no
+    // new render is triggered.
+    if (Object.values(Object.values(props.data)[0]).length === props.count) {
+      setMark("");
+      return;
+    }
+    // Update mark to be a new timestamp frequently if data is not complete
+    const timerId = setInterval(() => {
```

Comment:
instead of the mark and timer, could this work:
- add a new `completedOverallScore` state in App component
- in `handleUpdateOverallScore`,  check if scores are complete. If complete, update `completedOverallScore`
- pass `completedOverallScore` to `OverallScoreTable`

Reply:
This is a really good suggestion. Works perfect.